tests/unit/factored_matrix/test_properties.py

```diff
@@ -0,0 +1,174 @@
+            a, b = random_matrices[i]
+            expected_AB = a @ b
+            assert torch.allclose(factored_matrix.AB, expected_AB, atol=1e-5)
+
+    def test_BA_property(self, factored_matrices, random_matrices):
+        for i, factored_matrix in enumerate(factored_matrices):
+            a, b = random_matrices[i]
+            if a.shape[-2] == b.shape[-1]:
+                expected_BA = b @ a
+                assert torch.allclose(factored_matrix.BA, expected_BA, atol=1e-5)
+            else:
+                with pytest.raises(AssertionError):
+                    _ = factored_matrix.BA
+
+    def test_transpose_property(self, factored_matrices):
+        for factored_matrix in factored_matrices:
+            transposed_factored_matrix = factored_matrix.T
+            assert torch.allclose(
+                transposed_factored_matrix.A,
+                factored_matrix.B.transpose(-2, -1),
+                atol=1e-5,
+            )
+            assert torch.allclose(
+                transposed_factored_matrix.B,
+                factored_matrix.A.transpose(-2, -1),
+                atol=1e-5,
+            )
+
+    def test_svd_property(self, factored_matrices):
+        for factored_matrix in factored_matrices:
+            U, S, Vh = factored_matrix.svd()
+            assert torch.allclose(
+                factored_matrix.AB, U @ torch.diag_embed(S) @ Vh.T, atol=1e-5
+            )
+            # test that U and Vh are unitary
+            assert torch.allclose(U.T @ U, torch.eye(U.shape[-1]), atol=1e-5)
+            assert torch.allclose(Vh.T @ Vh, torch.eye(Vh.shape[-1]), atol=1e-5)
+
+    @pytest.mark.skip(
+        """
+        This test fails if there are leading ones. Talk to Neel about this.
+        """
+    )
+    def test_svd_property_leading_ones(self, factored_matrices_leading_ones):
+        for factored_matrix in factored_matrices_leading_ones:
+            U, S, Vh = factored_matrix.svd()
+            assert torch.allclose(
+                factored_matrix.AB, U @ torch.diag_embed(S) @ Vh.T, atol=1e-5
+            )
+            # test that U and Vh are unitary
+            assert torch.allclose(U.T @ U, torch.eye(U.shape[-1]), atol=1e-5)
+            assert torch.allclose(Vh.T @ Vh, torch.eye(Vh.shape[-1]), atol=1e-5)
+
+    @pytest.mark.skip(
+        """
+        Jaxtyping throws a TypeError when this test is run.
+        TypeError: type of the return value must be jaxtyping.Float[Tensor, '*leading_dims mdim']; got torch.Tensor instead
+
+        I'm not sure why. The error is not very informative. When debugging the shape was equal to mdim, and *leading_dims should
+        match zero or more leading dims according to the [docs](https://github.com/google/jaxtyping/blob/main/API.md).
+
+        Sort of related to https://github.com/neelnanda-io/TransformerLens/issues/190 because jaxtyping
+        is only enabled at test time and not runtime.
+        """
```

Comment:
@jbloomAus, I am not 100% sure what's going in here. Please can you advise?

Reply:
@rusheb I can't reproduce this? Did I write this or did you?

Reply:
I wrote it. Have you tried to unskip and run the entire test suite? 

I think the issue doesn't occur if the test is run in isolation, because multi arg checks won't be enabled